inventory/nodes_test.go

```diff
@@ -1,19 +1,326 @@
 package inventory
 
 import (
+	"context"
 	"testing"
 
 	"github.com/percona/pmm/api/inventory/json/client"
-	"github.com/stretchr/testify/assert"
+	"github.com/percona/pmm/api/inventory/json/client/nodes"
 	"github.com/stretchr/testify/require"
 
 	_ "github.com/Percona-Lab/pmm-api-tests" // init default client
 )
 
 func TestNodes(t *testing.T) {
 	t.Run("List", func(t *testing.T) {
+		remoteNodeID := addRemoteNode(t, "Test Remote Node for List")
+		defer removeNodes(t, remoteNodeID)
+
 		res, err := client.Default.Nodes.ListNodes(nil)
 		require.NoError(t, err)
-		assert.Len(t, res.Payload.Generic, 1)
+		require.NotZerof(t, len(res.Payload.Generic), "There should be at least one node")
+		require.Condition(t, func() (success bool) {
+			for _, v := range res.Payload.Generic {
+				if v.NodeID == "pmm-server" {
+					return true
+				}
+			}
+			return false
+		}, "There should be generic node with id `pmm-server`")
+		require.NotZerof(t, len(res.Payload.Remote), "There should be at least one node")
+		require.Conditionf(t, func() (success bool) {
+			for _, v := range res.Payload.Remote {
+				if v.NodeID == remoteNodeID {
+					return true
+				}
+			}
+			return false
+		}, "There should be remote node with id `%s`", remoteNodeID)
+	})
+}
+
+func TestGetNode(t *testing.T) {
+	t.Run("Basic", func(t *testing.T) {
+		params := &nodes.GetNodeParams{
+			Body:    nodes.GetNodeBody{NodeID: "pmm-server"},
+			Context: context.TODO(),
+		}
+		res, err := client.Default.Nodes.GetNode(params)
+		require.NoError(t, err)
+		require.NotNil(t, res.Payload.Generic)
+		require.Equal(t, res.Payload.Generic.NodeID, "pmm-server")
+		require.Equal(t, res.Payload.Generic.NodeName, "PMM Server")
+		require.Nil(t, res.Payload.Container)
+		require.Nil(t, res.Payload.Remote)
+		require.Nil(t, res.Payload.RemoteAmazonRDS)
+	})
+
+	t.Run("NotFound", func(t *testing.T) {
+		params := &nodes.GetNodeParams{
+			Body:    nodes.GetNodeBody{NodeID: "pmm-not-found"},
+			Context: context.TODO(),
+		}
+		res, err := client.Default.Nodes.GetNode(params)
+		require.Error(t, err) // Can't use EqualError because it returns different references each time.
```

Comment:
No, we should do better. The returned error is typed – it is a struct with specific fields. We should check them.